### PR TITLE
README Tests section improvements

### PR DESCRIPTION
**Pull request by eyalbe4**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
- Added ```-timeout 0``` options to the tests command.
- Modified the *Tests* section formating and structure.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,10 +154,24 @@ We welcome pull requests from the community.
 To run tests on the source code, you'll need a running JFrog Artifactory Pro instance.
```
Comment (yahavi) [inline README.md:154]: Artifactory Pro for Artifactory tests
Xray for the Xray tests
Distribution and an Enterprise+ license for Distribution tests

File: README.md [hunk 2]
```diff
@@ -154,10 +154,24 @@ We welcome pull requests from the community.
...
+### Test Types
+| Type                | Description        |
+| ---                 | ---                |
+| `-test.artifactory` | Artifactory tests  |
+| `-test.distribution`| Distribution tests |
+| `-test.xray`        | Xray tests         |
+| `-test.pipelines`   | Pipelines tests    |
+| `-test.access`      | Access tests       |
 
+### Flags
```
Comment (yahavi) [inline README.md:174]: To avoid confusion with the Go flags
```suggestion
### Authentication Flags
```

File: README.md [hunk 3]
```diff
@@ -177,19 +191,6 @@ Optional flags:
 | `-access.url`        | [Optional] JFrog Access URL.                                                                           |
```
Comment (yahavi) [inline README.md:191]: I suggest moving the `-access.url` up, to be with the other URLs.

File: README.md [hunk 4]
```diff
@@ -154,10 +154,24 @@ We welcome pull requests from the community.
 To run tests on the source code, you'll need a running JFrog Artifactory Pro instance.
 Use the following command with the below options to run the tests.
 ```sh
-go test -v github.com/jfrog/jfrog-client-go/tests [test-types] [flags]
+go test -v github.com/jfrog/jfrog-client-go/tests -timeout 0 [test-types] [flags]
 ```
-Optional flags:
+If you'd like to run a specific test, add the test function name using the ```-run``` flag. For example:
+```sh
+go test -v github.com/jfrog/jfrog-client-go/tests -timeout 0 -run TestGetArtifactoryVersionWithCustomHttpClient [test-types] [flags]
```
Comment (yahavi) [inline README.md:161]:
```suggestion
go test -v github.com/jfrog/jfrog-client-go/tests -timeout 0 -run TestGetArtifactoryVersionWithCustomHttpClient -test.artifactory -rt.url=http://127.0.0.1:8081/artifactory -rt.user=admin -rt.password=password
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Comment by eyalbe4** [comment]
Great feedback @yahavi. Implemented all.

**Pull request merged**